src/utils/hooks/use-generate-pathname.js

```diff
@@ -0,0 +1,49 @@
+import { useCallback } from 'react';
+import { useHistory, generatePath } from 'react-router-dom';
+import { routes } from '../../routes.config';
+import { localStorageName } from '../../config';
+
+/**
+ * To generate different pathnames based on each action
+ * E.g.: click on a node, or focus on a modular pipeline
+ * or to reset the pathname to the main page
+ */
+export const useGeneratePathname = () => {
+  const history = useHistory();
+
+  const localStorage = window.localStorage.getItem(localStorageName);
+  const activePipeline = JSON.parse(localStorage)?.pipeline?.active;
```

Comment:
I think you need to read from the URL here first, and if the URL isn't there then you could also get this from `state.pipeline.active` if you want to. 

Rashida and I were passing around URLs to each other while reviewing and we were seeing that if my `localStorage` pipeline value didn't match what was in her sent URL then the behavior wouldn't work as we intend it. That's why I think we need to read from the URL first, and then move on to what's on the user's machine.

If you use Redux instead of `localStorage`, you'd probably have to pass that as an argument into your hook and then use it here. I don't see any issue with `localStorage` either though. They'll always be in sync.

Reply:
ah yeah good point, i didnt think about that scenario, will update this 👍

Reply:
hmm actually after looking at this again, it's a bit unclear to me. Maybe best we can sync up and you can show me this case in more details? @tynandebold